packages/tracing/src/transaction.ts

```diff
@@ -105,6 +109,18 @@ export class Transaction extends SpanClass implements TransactionInterface {
     this.spanRecorder.add(this);
   }
 
+  /**
+   * @inheritDoc
+   */
+  public setContext(key: string, context: Context | null): void {
+    if (context === null) {
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      delete this._contexts[key];
+    } else {
+      this._contexts = { ...this._contexts, [key]: context };
```

Comment:
l: Is it important to make this a new object every time? Or could we just do:

```js
this._contexts[key] = context;
```

? To avoid re-creating objects when we don't really need to do so.

Reply:
I copied this from https://github.com/getsentry/sentry-javascript/blob/351be067ac7ff280155c128011f79dac9caf6475/packages/core/src/scope.ts#L257-L270

I agree though, will change. Perhaps we can update the scope file as well then?

Reply:
I think that makes sense as well 👍

Reply:
[536dca6](https://github.com/getsentry/sentry-javascript/pull/6154/commits/536dca6f03fcd11d16385ab85c0764a5270c9991)